spec/system/candidate_interface/carry_over/candidate_carries_over_before_apply_opens_spec.rb

```diff
@@ -0,0 +1,90 @@
+require 'rails_helper'
+
+RSpec.describe 'Carry after find opens but before apply opens' do
+  include CandidateHelper
+
+  scenario 'candidate carries over application', time: mid_cycle(RecruitmentCycle.current_year) do
+    given_i_have_an_unsubmitted_application
+    and_find_has_opened_but_apply_has_not
+    and_a_course_exists
+
+    when_i_sign_in
+    and_carry_over_my_application
+    and_complete_my_details
+    then_i_can_add_courses
+    and_i_cannot_submit_my_applications
+  end
+
+private
+
+  def given_i_have_an_unsubmitted_application
+    @candidate = create(:candidate)
+    create(:application_form, :unsubmitted, :with_completed_references, :eligible_for_free_school_meals, candidate: @candidate)
+  end
+
+  def and_find_has_opened_but_apply_has_not
+    advance_time_to(after_find_opens(RecruitmentCycle.next_year))
+  end
+
+  def and_a_course_exists
+    @course = create(:course, :open)
+    create(:course_option, course: @course)
+  end
+
+  def when_i_sign_in
+    login_as @candidate
+    visit root_path
+  end
+
+  def and_carry_over_my_application
+    click_on 'Update your details'
+  end
+
+  def and_complete_my_details
+    complete_the_equality_and_diversity_section
+    complete_the_references_section
+  end
+
+  def complete_the_equality_and_diversity_section
```

Comment:
Is this available in the `CandidateHelper`?

Reply:
Yes! It is, I hadn't looked at it closely before. Thanks!